libraries/chain/controller.cpp

```diff
@@ -3878,11 +3878,13 @@ struct controller_impl {
           });
    }
 
+   enum class full_qc_validation_t { no, yes };
+
    // Verify QC claim made by finality_extension in header extension
    // and quorum_certificate_extension in block extension are valid.
    // Called from net-threads. It is thread safe as signed_block is never modified after creation.
    // -----------------------------------------------------------------------------
-   void verify_proper_block_exts( const block_id_type& id, const signed_block_ptr& b, const block_header_state& prev ) {
+   void verify_proper_block_exts( const std::optional<block_id_type>& id, const signed_block_ptr& b, const block_header_state& prev, full_qc_validation_t qc_validation ) {
```

Comment:
I think you should not pass in `id` and have this method (and the others) return `qc_claim_t`. Then log from `create_block_handle_i` if not a `legacy_block`.

Reply:
Thanks. This will avoid the misleading logging when replaying.